console/module/api/src/main/java/org/eclipse/kapua/app/console/module/api/server/util/KapuaExceptionHandler.java

```diff
@@ -131,6 +131,10 @@ public static void handle(Throwable t) throws GwtKapuaException {
             throw new GwtKapuaException(GwtKapuaErrorCode.MAX_NUMBER_OF_ITEMS_REACHED, t, ((KapuaMaxNumberOfItemsReachedException) t).getArgValue());
         } else if (t instanceof DeviceMenagementException) {
             throw new GwtKapuaException(GwtKapuaErrorCode.valueOf(((DeviceMenagementException)t).getCode().name()), t, t.getLocalizedMessage());
+        } else if (t instanceof KapuaException && ((KapuaException) t).getCode().name().equals(KapuaErrorCodes.SAME_START_AND_DATE.name())){
```

Comment:
Why is check made on name of enum and not enum itself. Same for other if statements.

Reply:
Because, when I have enum itself, if clause is not executed. Then I saw that for KapuaErrorCodes.DUPLICATE_NAME  enum check is made with enum name and it works.